plugins.py

```diff
@@ -1,34 +1,64 @@
 #!/bin/env python
 
 import os
-from shutil import copy
+import cgi
+import urllib.request
+import random
+import string
+from shutil import copyfileobj
 from distutils.dir_util import copy_tree
 from distutils import log
 
-pluginsKey = "SSO_PLUGINS"
-pluginsVolume = "/opt/plugins"
+extensionsKey = "KEYCLOAK_EXTENSIONS"
+extensionsVolume = "/opt/extensions"
 themesPath = "/opt/themes"
 
-def getPlugins():
-    if "SSO_PLUGINS" in os.environ:
-        return os.environ[pluginsKey].split(",")
-    else:
-        return []
+def randomFileName():
+    letters = string.ascii_lowercase
+    return ''.join(random.choice(letters) for i in range(10)) + ".jar"
 
-def copyIfExists(plugin):
-    pluginPath = './' + plugin + '.jar'
-    if os.path.isfile(pluginPath):
-        print("Copying plugin: " + plugin)
-        copy(pluginPath, pluginsVolume)
+# Try to get the filename from the response headers and return
+# a random name if that fails
+def parseHeader(headers):
+    header = headers["Content-Disposition"]
+    if not header:
+        return randomFileName()
```

Comment:
Could you explain why if this header doesn't exist that a random filename will be generated? Does this header always contain the file name?

Reply:
All file downloads should populate the `Content-Disposition` header. This is just an extra measure to make sure we can fetch the file even if the header is missing for some reason.

Reply:
sounds good